packages/builder/src/registry.ts

```diff
@@ -77,18 +76,16 @@ export class CannonRegistry {
     );
   }
 
-  readIpfs(urlOrHash: string): Promise<Buffer> {
+  async readIpfs(urlOrHash: string): Promise<Buffer> {
     const hash = urlOrHash.replace(/^ipfs:\/\//, '');
 
-    return new Promise((resolve, reject) => {
-      const bufs: Buffer[] = [];
+    const bufs: Uint8Array[] = [];
 
-      const readable = Readable.from(this.ipfs.get(hash));
+    for await (const chunk of this.ipfs.cat(hash)) {
```

Comment:
Didn't remember we had .cat before 😬

But, added this fix here: https://github.com/usecannon/cannon/pull/26/files because the `.get` method was prefixing the buffer with some header data, so it couldn't be parsed. And the `.cat` method just returned the content in the case I was trying it out.